insights/parsers/lsinitrd.py

```diff
@@ -0,0 +1,90 @@
+"""
+LSINITRD - command ``lsinitrd``
+===============================
+
+This parser parses the filtered output of command ``lsinitrd`` and provides the
+info of listed files.
+"""
+
+from insights import parser, CommandParser
+from insights.core.ls_parser import Directory
+from insights.specs import Specs
+from insights.parsers import keyword_search
+
+
+@parser(Specs.lsinitrd)
+class LSINITRD(CommandParser):
+    """
+    A parser for command "lsinitrd".
+
+    Attributes:
+        data (dict): The key is the filename, the value is a dict describe
+            the file's info.
+        unparsed_lines(list): List of strings for unparsed lines.
+
+    As this lsinitrd spec is set to filterable, the structure of the output
+    is broken. Hence, this parser will parse only filelisting like lines in
+    output of 'lisinitrd', and also store all the unparsed lines.
+    If the other parts of the output structure are required in the future,
+    an enhancement may be performed then.
+
+    Examples:
+        >>> len(ls.data)
+        5
+        >>> assert ls.search(name__contains='kernel') == [
+        ...    {'group': 'root', 'name': 'kernel/x86', 'links': 3, 'perms': 'rwxr-xr-x',
+        ...     'raw_entry': 'drwxr-xr-x   3 root     root            0 Apr 20 15:58 kernel/x86',
+        ...     'owner': 'root', 'date': 'Apr 20 15:58', 'type': 'd', 'dir': '', 'size': 0}
+        ... ]
+        >>> "udev-rules" in ls.unparsed_lines
+        True
+    """
+
+    def parse_content(self, content):
+        file_types = set(['s', 'd', 'p', 'l', '-', 'c', 'b'])
+        perm = set(['-w-', 'rw-', '--x', '-wx', '---', 'rwx', 'r--', 'r-x'])
+        entries = []
+        _unparsed_lines = []
+
+        for l in content:
+            if l and l[0] in file_types and l[1:4] in perm:
+                entries.append(l)
+            else:
+                _unparsed_lines.append(l)
+
+        d = Directory("", len(entries), entries)
```

Comment:
Better use the `insights.core.ls_parser.parse` instead of the `insights.core.ls_parser.Directory`, and raise a `Exception` when nothing to parse.

```python
if entries:
    self.data =  parse(entries, 'lsinitrd').get('lsinitrd')['entries']
else:
    raise SkipException
```

Reply:
`insights.core.ls_parser.parse` is used.